assimilation_code/modules/assimilation/filter_mod.f90

```diff
@@ -2719,6 +2730,50 @@ subroutine set_copies(cnum, STAGE_COPIES)
 
 end subroutine set_copies
 
+
+!-------------------------------------------------------------------------------
+!> The infl_flavor namelist is a string, and specifies the inflation algorithm.
+!> The character string can either be the name associated with the type of inflation
+!> or the integer associated with the type of inflation. The string names of the
+!> inflation algorithms is based on what is declared in the adaptive_inflate_mod.f90
+!> which is repeated here for reference. 
+!>
+!> NO_INFLATION               = 0
+!> OBS_INFLATION              = 1    observation-space inflation (deprecated)
+!> VARYING_SS_INFLATION       = 2    spatially-varying state-space inflation
+!> SINGLE_SS_INFLATION        = 3    spatially-constant state-space inflation
+!> RELAXATION_TO_PRIOR_SPREAD = 4    (available only with posterior inflation)
+!> ENHANCED_SS_INFLATION      = 5    Inverse Gamma version of VARYING_SS_INFLATION
+
+function set_inflation_flavor(flavor_string) result(flavors)
+
+character(len=*), intent(in)  :: flavor_string(2)
+integer                       :: flavors(2)
+
+integer :: int_options(7) = (/ NO_INFLATION,               &
+                               OBS_INFLATION,              &
+                               VARYING_SS_INFLATION,       &
+                               SINGLE_SS_INFLATION,        &
+                               RELAXATION_TO_PRIOR_SPREAD, &
+                               RELAXATION_TO_PRIOR_SPREAD, &
+                               ENHANCED_SS_INFLATION       /)
+
+character(len=32) :: string_options(7) = (/ 'NO_INFLATION              ',&
+                                            'OBS_INFLATION             ',&
+                                            'VARYING_SS_INFLATION      ',&
+                                            'SINGLE_SS_INFLATION       ',&
+                                            'RELAXATION_TO_PRIOR_SPREAD',&
+                                            'RTPS                      ',&
+                                            'ENHANCED_SS_INFLATION     ' /)
+
+flavors(PRIOR)     = flex_parser(flavor_string(PRIOR),     &
+                          int_options, string_options, 'input_nml:inf_flavor(1)')
+flavors(POSTERIOR) = flex_parser(flavor_string(POSTERIOR), &
+                          int_options, string_options, 'input_nml:inf_flavor(2)')
+
+end function set_inflation_flavor
```

Comment:
this looks good - the calling code isn't too messy and now you can use the same code for all the places where integers should be strings.